executor/runtime/docker/docker_linux.go

```diff
@@ -116,37 +147,85 @@ func setupSystemPods(parentCtx context.Context, c *runtimeTypes.Container, cfg c
 	defer conn.Close()
 
 	// TODO: Make concurrent
-	if err := startSystemdUnit(ctx, conn, false, c.TaskID, atlasSystemdUnit); err != nil {
-		logrus.WithError(err).Error("Error starting atlas systemd unit")
+	if err := startSystemdUnit(ctx, conn, c.TaskID, c.ID, atlasServiceOpts); err != nil {
+		logrus.WithError(err).Error("Error starting atlas service")
 		return err
 	}
 	if cfg.ContainerSSHD {
-		if err := startSystemdUnit(ctx, conn, true, c.TaskID, sshdSystemdUnit); err != nil {
-			logrus.WithError(err).Error("Error starting ssh systemd unit")
+		if err := startSystemdUnit(ctx, conn, c.TaskID, c.ID, sshServiceOpts); err != nil {
+			logrus.WithError(err).Error("Error starting ssh service")
 			return err
 		}
 	}
-	if err := startSystemdUnit(ctx, conn, true, c.TaskID, metadataServiceSystemdUnit); err != nil {
-		logrus.WithError(err).Error("Error starting metadata service systemd unit")
+	if err := startSystemdUnit(ctx, conn, c.TaskID, c.ID, metadataServiceOpts); err != nil {
+		logrus.WithError(err).Error("Error starting metadata service")
 		return err
 	}
 
 	if shouldStartMetatronSync(&cfg, c) {
 		// The metatron sync service queries the metadata server, so it needs to be started after
-		if err := startSystemdUnit(ctx, conn, true, c.TaskID, metatronServiceSystemdUnit); err != nil {
-			logrus.WithError(err).Error("Error starting metatron systemd unit")
+		if err := startSystemdUnit(ctx, conn, c.TaskID, c.ID, metatronServiceOpts); err != nil {
+			logrus.WithError(err).Error("Error starting metatron service")
 			return err
 		}
 	}
 
 	return nil
 }
 
-func startSystemdUnit(ctx context.Context, conn *dbus.Conn, required bool, taskID, unitName string) error {
-	qualifiedUnitName := fmt.Sprintf("%s@%s.service", unitName, taskID)
-	l := logrus.WithField("taskID", taskID).WithField("unit", unitName)
-	l.Infof("Starting systemd unit %s", qualifiedUnitName)
+func runServiceInitCommand(ctx context.Context, log *logrus.Entry, cID string, opts serviceOpts) error {
+	if opts.initCommand == "" {
+		return nil
+	}
+
+	var stdout bytes.Buffer
```

Comment:
Why don't we combine sydout / stderr?

Reply:
2 reasons:

The first is that cmd.CombinedOutput doesn't accept a Context, and I want the process to be killed when the executor times out.

The second is that I'd like this "contract" with service owners to be as explicit and deterministic as possible: I don't want the message that gets bubbled up to users to change if the service starts printing a statement to stderr that it didn't previously.